native/acter/src/api/room.rs

```diff
@@ -522,6 +524,85 @@ impl Room {
             .await?
     }
 
+    pub async fn add_parent_room(&self, room_id: String, canonical: bool) -> Result<String> {
+        let room_id = OwnedRoomId::try_from(room_id)?;
+        if !self
+            .get_my_membership()
+            .await?
+            .can(crate::MemberPermission::CanLinkSpaces)
+        {
+            bail!("You don't have permissions to add parent-rooms");
+        }
+        if !self.is_joined() {
+            bail!("You can't update a room you aren't part of");
+        }
+        let room = self.room.clone();
+        let client = self.core.client().clone();
+
+        RUNTIME
+            .spawn(async move {
+                let Some(Ok(homeserver)) = client.homeserver().host_str().map(|h| h.try_into()) else {
+                    return Err(Error::HomeserverMissesHostname)?;
+                };
+                let mut content = SpaceParentEventContent::new(vec![homeserver]);
+                if canonical {
+                    content.canonical = true;
+                }
```

Comment:
unnecessary check.
```suggestion
                content.canonical = canonical;
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1089/commits/c4d76b5ceb55983bb2c1a78e77194d0fc91f9d6c.
I just wanted to avoid unnecessary assignment when `canonical` is `false`.